vnext/Shared/DevSupportManager.cpp

```diff
@@ -47,7 +47,32 @@ std::future<std::pair<std::string, bool>> DownloadFromAsync(const std::string &u
   co_await winrt::resume_background();
 
   winrt::Windows::Web::Http::HttpRequestMessage request(winrt::Windows::Web::Http::HttpMethod::Get(), uri);
-  winrt::Windows::Web::Http::HttpResponseMessage response = co_await httpClient.SendRequestAsync(request);
+  auto asyncRequest = httpClient.SendRequestAsync(request);
+#ifdef DEFAULT_CPPWINRT_EXCEPTIONS
+  try {
+    winrt::Windows::Web::Http::HttpResponseMessage response = co_await asyncRequest;
+  } catch (winrt::hresult_error const &e) {
+    co_return std::make_pair(
+        Microsoft::Common::Unicode::Utf16ToUtf8(e.message().c_str(), e.message().size()).c_str(), false);
+  }
+#else
+  co_await lessthrow_await_adapter<winrt::Windows::Foundation::IAsyncOperationWithProgress<
+      winrt::Windows::Web::Http::HttpResponseMessage,
+      winrt::Windows::Web::Http::HttpProgress>>{asyncRequest};
+
+  HRESULT hr = asyncRequest.ErrorCode();
+  if (FAILED(hr)) {
+    std::ostringstream sstream;
+    if (hr == WININET_E_CANNOT_CONNECT) {
+      sstream << "A connection with the server " << url << " could not be established\n\nIs the packager running?";
+    } else {
+      sstream << "Error " << std::hex << static_cast<int>(asyncRequest.ErrorCode()) << " downloading " << url;
+    }
+    co_return std::make_pair(sstream.str(), false);
```

Comment:
FYI I think as of C++ 17, now that we have CTAD, make_tuple, make_pair, etc is unneeded since they were just helpers to automatically infer type. 

Same above.